ext/resourcemanager/optimus.go

```diff
@@ -33,16 +35,43 @@ func NewOptimusResourceManager(resourceManagerConfig config.ResourceManager) (*O
 	if err := mapstructure.Decode(resourceManagerConfig.Config, &conf); err != nil {
 		return nil, fmt.Errorf("error decoding resource manger config: %w", err)
 	}
+
 	if conf.Host == "" {
 		return nil, errors.New("optimus resource manager host is empty")
 	}
+
+	httpClient, err := newHTTPClient(conf.Host)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing http client: %w", err)
+	}
+
 	return &OptimusResourceManager{
 		name:       resourceManagerConfig.Name,
 		config:     conf,
-		httpClient: http.DefaultClient,
+		httpClient: httpClient,
 	}, nil
 }
 
+func newHTTPClient(host string) (*http.Client, error) {
+	httpClient := new(http.Client)
+
+	if strings.HasPrefix(host, "https") {
+		certPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("error reading system certificate: %w", err)
+		}
+
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:            certPool,
+				InsecureSkipVerify: true,
```

Comment:
Can we remove `InsecureSkipVerify` from tlsConfig.

Reply:
sure, we can do that

Reply:
updated as suggested